### PR TITLE
Samba4 (from prev. repository)

### LINKED ISSUE #1693: Update Request: Samba
The samba in Voids repos is still `3.6.25`, however the current version is `4.8.3`.

Right now some clients cannot access my shares anymore since SMB3 is required for up-to-date windows computers. 

https://github.com/void-linux/void-packages/blob/4e73c26f56504ee1ef1884204cf53a24c0bd21e9/srcpkgs/samba/template

### PR DESCRIPTION
**Pull request by fosslinux**
This was a pull request in the previous repository (https://github.com/voidlinux/void-packages/pull/6601). If we can iron out any bugs and get this merged it would be great.

Closes #1693.

/cc @Gottox you made this pull request in the old repository.

Also a new maintainer is needed as Juan is now gone...

### CODE REVIEW

**Comment by maxice8** [comment]
there are conflicts in common/shlibs, please fix them, i would put the samba4 libs directly below the samba3 libs

**Comment by fosslinux** [comment]
@maxice8 this is not my repository, it is @Gottox 's , so I can't fix them.

**Comment by Gottox** [comment]
Didn't knew that you could even open PRs from repositories you don't own.

**Comment by fosslinux** [comment]
@Gottox interesting feature I found when I was helping to revive a dead project, there were a whole lot of useful features we wanted in the fork, and I noticed 'head fork' and saw a huge list. kind of like void linux is now 😃

**Comment by Gottox** [comment]
shame on me :laughing:

File: srcpkgs/heimdal/template [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Template file for 'heimdal'
+pkgname=heimdal
+version=7.1.0
+revision=1
+build_style=gnu-configure
+makedepends="ncurses-devel"
+short_desc="Kerberos 5, PKIX, GSS-API, SPNEGO, NTLM, MD5, SASL implementation"
+maintainer="Enno Boland <gottox@voidlinux.eu>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/heimdal/template]: please use SPDX identifier

File: srcpkgs/samba4/template [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Template file for 'samba'
+pkgname=samba4
+version=4.8.2
+revision=1
+build_style=waf
+configure_script=buildtools/bin/waf
+configure_args="--sysconfdir=/etc/samba \
```
Comment (maxice8) [inline srcpkgs/samba4/template]: no need for \ at the end of lines, you're in a multi-line string

File: srcpkgs/samba4/template [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ --with-ads \
+ --with-shared-modules=idmap_ad,idmap_rid,idmap_adex,idmap_hash,idmap_tdb2,pdb_tdbsam,pdb_ldap,pdb_ads,pdb_smbpasswd,pdb_wbc_sam,pdb_samba4,auth_unix,auth_wbc,auth_server,auth_netlogond,auth_script,auth_samba4
+ --enable-cups \
+ --without-gettext
+"
+patch_args="-Np1"
+wrksrc="samba-$version"
+short_desc="SMB/CIFS file, print, and login server for Unix"
+maintainer="Enno Boland <gottox@voidlinux.eu>"
+license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/samba4/template]: ditto

**Comment by maxice8** [comment]
Please rebase and make the proposed changes.

**Comment by Gottox** [comment]
I currently have very little time to work on this. :disappointed:

**Comment by Anachron** [comment]
I'll try to work on this later. There are some windows machines in my network having issues with `samba < 4`.

**Comment by maxice8** [comment]
Please re-open under your own branch with the changes proposed.

**Pull request closed without merging**